### PR TITLE
Add Exceptions Input

### PR DESCRIPTION
**Pull request by calsupik**
Ability to pass in multi line string of schedule names to be excluded from the revision update

### CODE REVIEW

**Review by wrightmalone — APPROVED** [review]

**Review by zachfey — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -14,13 +14,18 @@ const {
...
 
     console.log(`Getting schedules in the group ${groupName}.`);
-    const { Schedules: schedules } = await scheduler.send(
+    let { Schedules: schedules } = await scheduler.send(
       new ListSchedulesCommand({ GroupName: groupName })
     );
 
+    schedules = schedules.filter(
+      (schedule) => !exceptions.includes(schedule.Name)
+    );
```
Comment (calsupik) [inline index.js:27]: @zachfey This is where we are filtering out any exceptions passed to not update.

File: index.js [hunk 2]
```diff
@@ -14,13 +14,18 @@ const {
     const region = core.getInput('region')
     const groupName = core.getInput('group_name')
     const newTaskDefinitionArn = core.getInput('new_task_definition_arn')
+    const exceptions = core.getMultilineInput('exceptions') || [];
```
Comment (calsupik) [inline index.js:17]: The `getMultilineInput` allows us to get the string list of exceptions that is separated by newline.

**Comment by calsupik** [comment]
@zachfey Not sure what has to happen to "publish" the action after we merge into `main`, and maybe we need to update the version being called in our `deploy.yml` file where its being called.

**Comment by zachfey** [comment]
> @zachfey Not sure what has to happen to "publish" the action after we merge into `main`, and maybe we need to update the version being called in our `deploy.yml` file where its being called.

our `deploy.yml` references this repo via `risk3sixty/update-scheduler-task-definition-revisions@main` so it should automatically use what's in main.  I can't remember how to release to the marketplace off the top of my head

**Pull request merged**